axon-server-connector/src/main/java/org/axonframework/axonserver/connector/heartbeat/Scheduler.java

```diff
@@ -0,0 +1,100 @@
+package org.axonframework.axonserver.connector.heartbeat;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+/**
+ * Responsible to schedule generic tasks at a fixed rate.
+ * The scheduling can be started an stopped at any time.
+ *
+ * @author Sara Pellegrini
+ * @since 4.2.1
+ */
+public class Scheduler {
```

Comment:
Why not using `SchedulerExecutorService` only instead?

I can imagine a situation where you want to mock this in tests. In that case, I'd build a wrapper around `SchedulerExecutorService`.

Reply:
Interesting question, which I'd like to hear the answer for too.

Reply:
I've replaced it with an interface that allows me to properly test HeartbeatMonitor